dvjsharma/Drawn2Shoe
# fix: Implemented Account verification , Email validation. issue:#152

## Description

This PR implements email validation functionality in our application. The feature includes temporary user storage, token-based email verification, and user redirection upon successful verification. Below are the key details of the implementation:

### Changes

**1. Temporary User Storage:**

  - When a new user signs up, their details are stored in a temporary database table (tempUser).
  - This ensures that only verified users are moved to the main user database (mainuser).

**2. Token Generation and Email Sending:**

  - A unique token is generated for each user upon signup.
  - An email with a verification link containing the token and user's email is sent to the user.

**3. Email Verification Endpoint:**

  - An API endpoint `(/api/users/verify-email)` is created to handle the verification process.
  - The endpoint validates the token and email, transfers the user details to the main user database if successful, and deletes the temporary entry.

**4. Frontend Integration:**

  - The `VerifyEmail` component fetches the token and email from the URL and calls the verification API.
  - On successful verification, the user is redirected to a success page with a "Login" button that redirects to the login page.

![image](https://github.com/user-attachments/assets/1d195ad9-fe8f-4e03-8224-2027207e3ea2)
![image](https://github.com/user-attachments/assets/d984654f-1e00-415f-be96-3fbac35e2f7d)
![image](https://github.com/user-attachments/assets/de53bb48-cd95-4c61-8622-8a7722a2964c)


## How to test

- Ensure you have the backend server running on port 3000 and the frontend application running on port 5173.
- Set up your email service correctly to send verification emails.
- Make sure your Prisma Client is properly configured and the database is up and running.
- Go to the signup page.
- Fill in details (name, email, password, etc.).
- Submit the form.
- Check your email for a verification link.
- Click the verification link in the email.
- Verify you see "Your account has been verified successfully!" on the /verify-email page.
- Ensure user details are in tempUser table initially.
- After verification, check details are moved to mainuser table and removed from tempUser.


fix an issue already created in the repository-
#152
